projects/packages/publicize/tests/php/test-connections-post-field.php

```diff
@@ -336,4 +336,25 @@ public function test_update_connections_by_service_name() {
 			$this->assertSame( 'facebook' !== $connection->service_name, $connection->enabled );
 		}
 	}
+
+	/**
+	 * Test that connections are skipped when the publicize_checkbox_default filter is used.
+	 */
+	public function test_default_checkbox_filter() {
+		// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$filter_func = function ( $default ) {
+			return false;
+		};
+
+		add_filter( 'publicize_checkbox_default', $filter_func );
+		$request = new WP_REST_Request( 'POST', sprintf( '/wp/v2/posts/%d', $this->draft_id ) );
+		$this->server->dispatch( $request );
+
+		foreach ( self::$connection_ids as $unique_id ) {
+			$skip_key = $this->publicize->POST_SKIP . $unique_id;
+			$this->assertNotEmpty( get_post_meta( $this->draft_id, $skip_key, true ) );
+		}
+
+		remove_filter( 'publicize_checkbox_default', $filter_func );
```

Comment:
Does it make sense to add an extra test testing the opposite (if default filter is not used, connections are enabled by default)?

Reply:
Good idea. Added in 0728743435